akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1352,6 +1352,31 @@ trait FlowOps[+Out, +Mat] {
     }
 
   /**
+   * Merge the given [[Source]] to this [[Flow]], taking elements as they arrive from input streams,
+   * picking always the smallest of the available elements (waiting for one element from each side
+   * to be available).
+   *
+   * '''Emits when''' one of the inputs has an element available
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' all upstreams complete
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def mergeSorted[U >: Out, M](that: Graph[SourceShape[U], M])(implicit ord: Ordering[U]): Repr[U] =
+    via(mergeSortedGraph(that))
+
+  protected def mergeSortedGraph[U >: Out, M](that: Graph[SourceShape[U], M])(implicit ord: Ordering[U]): Graph[FlowShape[Out @uncheckedVariance, U], M] =
+    GraphDSL.create(that) { implicit b ⇒
```

Comment:
Given how many methods we now have with almost exactly this piece of code/scaffolding I would propose to attempt some reuse :)